code_generator/GdkCodeGenerator/src/Generation/SingleGenerationJob.cs

```diff
@@ -135,5 +134,31 @@ protected override void RunImpl()
                 Content.Add(Path.Combine(relativeOutputPath, referenceProviderFileName), referenceProviderTranslationCode);
             }
         }
+
+        /// <summary>
+        ///     Filters out auto-generated types like PositionData from the JSON AST.
+        ///     However, we want to keep types that are used as a "data" field in a component.
+        /// </summary>
+        private void SelectTypesToGenerate(UnitySchemaFile schemaFile)
+        {
+            var componentDataTypes =
+                schemaFile.ComponentDefinitions.Select(component => component.RawDataDefinition);
+
+            // From inspection of the JSON AST you can observe that a type definition is auto-generated if the following
+            // conditions are true:
+            //     1. The FQN type names are the same .
+            //     2. The source references are the same.
+            // Using this information, we can effectively filter out auto-generated types.
+            var filteredTypes = schemaFile.TypeDefinitions.Where(type => componentDataTypes.All(componentData =>
+                type.QualifiedName != componentData.TypeName ||
+                !SourceReferenceEquals(type.SourceReference, componentData.sourceReference)));
+
+            typesToGenerate.AddRange(filteredTypes);
```

Comment:
why not return the list instead of modifying within method?

Reply:
as discussed, not a problem

Reply:
😬